src/d3d12/d3d12_renderer.cpp

```diff
@@ -145,7 +145,7 @@ namespace wr
 		m_texture_pools.push_back(CreateTexturePool());
 	}
 
-	CPUTextures D3D12RenderSystem::Render(std::shared_ptr<SceneGraph> const & scene_graph, FrameGraph & frame_graph)
+	CPUTextures D3D12RenderSystem::Render(SceneGraph& scene_graph, FrameGraph& frame_graph)

```

Comment:
Can't we make the frame_graph a const& as well?

Reply:
We can't, the scene graph also can't. The renderer updates both depending if there are changes,